webui/src/lib/remark-plugins/imageUriReplacer.ts

```diff
@@ -0,0 +1,33 @@
+import { visit } from "unist-util-visit";
+import type { Node } from "unist";
+import type { Root } from "mdast";
+import type { Plugin } from "unified";
+
+import { objects } from "../api";
+
+const imageUriReplacer: Plugin<[], Root> = () => async (tree) => {
+  const images: Array<Node & { url: string }> = [];
+  visit(tree, "image", (node: Node & { url: string }) => {
+    if (node.url.startsWith("lakefs://")) {
```

Comment:
What about relative paths? I think that's the more common case. Our readme for example:
```md
    <img src="docs/assets/img/logo_large.png"/>
```

Reply:
I'll add relative URL support, but I must make some assumptions.  
When the URL starts with `lakefs://`, the format is `lakefs://{repo}/{branch}/...{path}`  
Regarding relative URLs, I'll assume it's relative to the current repo and branch. WDYT?

Reply:
Added relative URL support + relevant tests

Reply:
I think you're assumptions are spot on.

Reply:
Just `ref` instead of `branch` 🙈

Reply:
👍🏻